core/shared/src/main/scala/zio/logging/LogGroupEquivalence.scala

```diff
@@ -0,0 +1,12 @@
+package zio.logging
+
+final case class LogGroupEquivalence[A](equivalent: (A, A) => Boolean)
```

Comment:
In order to put some force behind the word `Equivalence`, we could make it a law that `equivalent` is commutative: if `a1` is equivalent to `a2`, then `a2` is equivalent to `a1`.

On the other hand, if this would cause a problem for log groups, then we should change the name to `LogGroupRelation` andd the name of the function to `matchesGroup` rather than `equivalent`.